crates/proto/src/rr/rdata/cert.rs

```diff
@@ -0,0 +1,705 @@
+/// type code is 37.
+///
+///    1 1 1 1 1 1 1 1 1 1 2 2 2 2 2 2 2 2 2 2 3 3
+/// 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+/// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+/// |             type              |             key tag           |
+/// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+/// |   algorithm   |                                               /
+/// +---------------+            certificate or CRL                 /
+/// /                                                               /
+/// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-|
+/// ```
+#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+pub struct CERT {
+    cert_type: CertType,
+    key_tag: u16,
+    algorithm: Algorithm,
+    cert_data: Vec<u8>,
+}
+
+/// [RFC 4398, Storing Certificates in DNS, November 1987][rfc4398]
+/// https://tools.ietf.org/html/rfc4398#section-2.1
+///
+/// ```text
+/// 2.1.  Certificate Type Values
+///
+///    The following values are defined or reserved:
+///
+///          Value  Mnemonic  Certificate Type
+///          -----  --------  ----------------
+///              0            Reserved
+///              1  PKIX      X.509 as per PKIX
+///              2  SPKI      SPKI certificate
+///              3  PGP       OpenPGP packet
+///              4  IPKIX     The URL of an X.509 data object
+///              5  ISPKI     The URL of an SPKI certificate
+///              6  IPGP      The fingerprint and URL of an OpenPGP packet
+///              7  ACPKIX    Attribute Certificate
+///              8  IACPKIX   The URL of an Attribute Certificate
+///          9-252            Available for IANA assignment
+///            253  URI       URI private
+///            254  OID       OID private
+///            255            Reserved
+///      256-65279            Available for IANA assignment
+///    65280-65534            Experimental
+///          65535            Reserved
+///
+/// ```
+#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Copy)]
+pub enum CertType {
+    /// 0, 255, 65535            Reserved
+    Reserved,
+
+    /// 1  PKIX      X.509 as per PKIX
+    PKIX,
+
+    /// 2  SPKI      SPKI certificate
+    SPKI,
+
+    /// 3  PGP       OpenPGP packet
+    PGP,
+
+    /// 4  IPKIX     The URL of an X.509 data object
+    IPKIX,
+
+    /// 5  ISPKI     The URL of an SPKI certificate
+    ISPKI,
+
+    /// 6  IPGP      The fingerprint and URL of an OpenPGP packet
+    IPGP,
+
+    /// 7  ACPKIX    Attribute Certificate
+    ACPKIX,
+
+    /// 8  IACPKIX   The URL of an Attribute Certificate
+    IACPKIX,
+
+    /// 253  URI       URI private
+    URI,
+
+    /// 254  OID       OID private
+    OID,
+
+    /// 9-252, 256-65279            Available for IANA assignment
+    AvailableIANAAssignment(u16),
```

Comment:
This name seems a little verbose, suggest just calling it `Available` or maybe `Unassigned` (while retaining the comment that has more context).